io.openems.edge.evse.chargepoint.alpitronic/src/io/openems/edge/evcs/alpitronic/EvcsAlpitronicImpl.java

```diff
@@ -188,35 +214,126 @@ public void handleEvent(Event event) {
 		}
 	}
 
+	/**
+	 * Defines the Modbus protocol for communication with Alpitronic Hypercharger.
+	 * 
+	 * <p>
+	 * Register mapping based on Load Management Manual v2.5:
+	 * <ul>
+	 * <li>Station-level input registers: 0-48 (global information)</li>
+	 * <li>Connector-specific input registers: 100+, 200+, 300+, 400+ (per connector
+	 * data)</li>
+	 * <li>Holding registers: for power control and reactive power settings</li>
+	 * </ul>
+	 * 
+	 * @return configured ModbusProtocol instance
+	 */
 	@Override
 	protected ModbusProtocol defineModbusProtocol() {
-		var modbusProtocol = new ModbusProtocol(this,
+		// Determine which protocol to use based on configured or detected version
```

Comment:
`defineModbusProtocol()` should always get called exactly once and only one `ModbusProtocol` should exist. Otherwise you will see problems with 'dangling' objects etc.

See Socomec implementation for an example on how to do fully dynamic protocol initialisation.